kafka/src/main/java/nl/nn/adapterframework/extensions/kafka/KafkaInternalListener.java

```diff
@@ -0,0 +1,202 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.extensions.kafka;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.IPullingListener;
+import nl.nn.adapterframework.core.ListenerException;
+import nl.nn.adapterframework.core.PipeLineResult;
+import nl.nn.adapterframework.core.PipeLineSession;
+import nl.nn.adapterframework.receivers.RawMessageWrapper;
+import nl.nn.adapterframework.stream.Message;
+import nl.nn.adapterframework.stream.MessageContext;
+import nl.nn.adapterframework.util.StringUtil;
+
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+/**
+ * This class should NOT be used outside of this kafka package.
+ * This class isn't public to prevent generation of javadoc/frankdoc.
+ * @param <T> Topic type.
+ * @param <M> Message type.
+ */
+class KafkaInternalListener<T,M> extends KafkaFacade implements IPullingListener<ConsumerRecord<T, M>> {
+	//setter is for testing purposes only.
+	private @Setter(AccessLevel.PACKAGE) Consumer<T, M> consumer;
+	private @Setter(AccessLevel.PACKAGE) Function<Properties, Consumer<T,M>> consumerGenerator = KafkaConsumer::new;
+	private Iterator<ConsumerRecord<T, M>> waiting;
+	private @Setter String groupId;
+	private @Setter boolean fromBeginning;
+	private @Setter int patternRecheckInterval;
+	private @Setter String topics;
```

Comment:
Those don't show up in the FrankDoc, because this is a Internal class. The user only sees the KafkaListener.

The class isn't public to prevent usage outside of the module. This also hides it from the FrankDoc/JavaDoc.